src/libical/icalenumarray.h

```diff
@@ -0,0 +1,55 @@
+/*======================================================================
+ FILE: icalenumarray.h
+
+ CREATOR: Ken Murchison 24 Aug 2022 <murch@fastmailteam.com>
+
+ SPDX-FileCopyrightText: 2022, Fastmail Pty. Ltd. (https://fastmail.com)
+
+ SPDX-License-Identifier: LGPL-2.1-only OR MPL-2.0
+
+ ======================================================================*/
+
+#ifndef ICALENUMARRAY_H
+#define ICALENUMARRAY_H
+
+#include "libical_ical_export.h"
+#include "icalarray.h"
+
+#include <stdlib.h>
+
+typedef icalarray icalenumarray;
+
+typedef struct {
+    int val;
+    const char *xvalue;
+} icalenumarray_element;
+
+#define icalenumarray_new(increment_size) \
+    icalarray_new(sizeof(icalenumarray_element), increment_size)
+
+#define icalenumarray_element_at(array, position) \
+    ((const icalenumarray_element *)icalarray_element_at(array, position))
+
+#define icalenumarray_size(array) ((array)->num_elements)
```

Comment:
we need to consider a dereference of a null pointer situation.
maybe:
`#define icalenumarray_size(array) (if(array) ? (array)->num_elements) : -1)` ?

same for the `vcardenumarray_size(array)` macro

Reply:
I'm all for being defensive against NULL pointer arguments, I just kept it as-is because none of the array functions, including icalarray are safe anyway. I'll submit a patch for that in this PR, too.